manifests/config.pp

```diff
@@ -1,35 +1,33 @@
 # Class icinga::config
 #
 # Configures icinga using the defaults set in the params class
-
-class icinga::config {
-
-  include icinga::params
-
-  $icinga_user = $icinga::params::icinga_user
-  $icinga_group = $icinga::params::icinga_group
-  $icinga_cmd_grp = $icinga::params::icinga_cmd_grp
-  $notifications = $icinga::params::notifications
-  $enable_ido = $icinga::params::enable_ido
-  $embedded_perl = $icinga::params::embedded_perl
-  $perfdata = $icinga::params::perfdata
-  $perfdatatype = $icinga::params::perfdatatype
-  $admin_group = $icinga::params::admin_group
-  $nagios_plugins = $icinga::params::nagios_plugins
-  $nagios_extra_plugins = $icinga::params::nagios_extra_plugins
-  $db_password = $icinga::params::db_password
-  $email_password = $icinga::params::email_password
-  $check_timeout = $icinga::params::check_timeout
-  $clickatell_api_id = hiera('monitoring::clickatell_api_id', undef)
-  $clickatell_username = hiera('monitoring::clickatell_username', undef)
-  $clickatell_password = hiera('monitoring::clickatell_password', undef)
-  $is_pbx = hiera('is_pbx', false)
-  $pbx_mngr_pw = hiera('monitoring::pbx_mngr_pw', undef)
-  $debug = $icinga::params::debug
-  $admin_email = $icinga::params::admin_email
-  $admin_pager = $icinga::params::admin_pager
-  $stalking = $icinga::params::stalking
-  $flap_detection = $icinga::params::flap_detection
+# should not be called directly
+class icinga::config (
+  $icinga_user          = $::icinga::icinga_user,
+  $icinga_group         = $::icinga::icinga_group,
+  $icinga_cmd_grp       = $::icinga::icinga_cmd_grp,
+  $notifications        = $::icinga::notifications,
+  $enable_ido           = $::icinga::enable_ido,
+  $embedded_perl        = $::icinga::embedded_perl,
+  $perfdata             = $::icinga::perfdata,
+  $perfdatatype         = $::icinga::perfdatatype,
+  $admin_group          = $::icinga::admin_group,
+  $nagios_plugins       = $::icinga::nagios_plugins,
+  $nagios_extra_plugins = $::icinga::nagios_extra_plugins,
+  $db_password          = $::icinga::db_password,
+  $email_password       = $::icinga::email_password,
+  $check_timeout        = $::icinga::check_timeout,
+  $clickatell_api_id    = $::icinga::clickatell_api_id,
+  $clickatell_username  = $::icinga::clickatell_username,
+  $clickatell_password  = $::icinga::clickatell_password,
+  $is_pbx               = $::icinga::is_pbx,
+  $pbx_mngr_pw          = $::icinga::pbx_mngr_pw,
+  $debug                = $::icinga::debug,
+  $admin_email          = $::icinga::admin_email,
+  $admin_pager          = $::icinga::admin_pager,
+  $stalking             = $::icinga::stalking,
+  $flap_detection       = $::icinga::flap_detection,
+) {
 
```

Comment:
Is there a reason you have put these in as class params?
I see these classes as private so they shouldn't have any parameters.
Their values should come in from the icinga class as you have here but without being parameters
Same would apply to the rest of the internal subclasses.